### ISSUE TITLE
Using columns from junction table in "where" clause

### ISSUE THREAD

**Issue by ratoi-crysty** [sequelize/sequelize#1775]
I've created a Many-to-Many relationship and I need to make a query using some columns from the junction table in "where" clause.

I have the following models:

```
sequelize.define("User", {
    //some values
}, {
    "classMethods" : {
        "associate" : function (models) {
            User.hasMany(models.Book, { "through" : models.BorrowedBook });
        }
    }
});

sequelize.define("Book", {
    //some values
}, {
    "classMethods" : {
        "associate" : function (models) {
            Book.hasMany(models.User, { "through" : models.BorrowedBook });
        }
    }
});

sequelize.define("BorrowedBook", {
        //some other values
        "returnedAt" : DataTypes.DATE
    }
);
```

I want to use in "where" clause a column from "BorrowedBook"?

E.g. I want to get all users with books eagerly loaded that have at least 1 book and "BorrowedBook.returnedAt" is null.

A feature to do this will be very helpful. Something like this:

```
db.Book.find({
    include: [
        { model: db.User, through: { where: {borrowedAt: null } }
    ]
})
```

**Comment by mickhansen** [comment]
Preliminary `through.where` support implemented in https://github.com/sequelize/sequelize/commit/4887b84030caedb4d85d71c2afe9030954dd374a.

Please test and open PR's with test cases for missing functionality.

**Comment by ericblade** [comment]
My apologies for resurrecting a long, long dead thread, but -- is this in the documentation somewhere?  I have been searching the intertubes for hours to figure out how to do this, and I finally stumbled upon this thread, which lead me to reading the chai test, which finally lead to me figuring out how to do this thing that i've been poking at for hours. :-)

**Comment by EduardsE** [comment]
@ericblade chai test?

**Comment by papb** [comment]
@EduardsE he probably means [this](https://github.com/sequelize/sequelize/commit/4887b84030caedb4d85d71c2afe9030954dd374a#diff-52a8641b8c0175da7b47275f77edce69)

**Comment by papb** [comment]
Closed by #11825, see [this section of the docs](https://sequelize.org/master/manual/eager-loading.html#eager-loading-with-many-to-many-relationships)
